src/components/product_copier/product_copier.requirements.yaml

```diff
@@ -0,0 +1,8 @@
+---
+description: a
+requirements:
+  - text: The component shall copy data products from one location to another, every time it is sent a Tick, given a list of source ID/destination ID mappings.
+  - text: The component shall fail at initialization when two mappings share the same destination.
+    description: When two mappings share the same destination, it's unclear which one should "win" and write to the destination first. If the user requires some kind of overwriting behavior, they can make this explicit by instantiating another of this component. (This is generally with the assumption that send connector is connected to some kind of database write operation.)
+  - text: If the component fetches an empty data product, it shall continue without errors and will not send anything to the destination.
```

Comment:
Agreed, the way it is currently written is unclear and doesn't really correspond to anything. However, I just noticed that Data_Product_Return can have a status of Success, Not_Available, or Id_Out_Of_Range. I'm thinking that it might be worth a few requirements to specify that Not_Available raises no errors and Id_Out_Of_Range should raise, any thoughts?